### PR TITLE
Error Reporting, Telemetry and User Feedback

### LINKED ISSUE #137: Error Reporting
**Describe the feature**
Error reporting to a remote service so we can be alerted of bugs in the application. This should also be extended to upload source maps for stack traces.

**Tasks**
- [x] User Feedback
- [x] Build integration for source maps
- [x] Error reporting

**⏰ Estimated Time**
4-5 hours

### PR DESCRIPTION
**Pull request by braunsonm**
This commit as Sentry for remote error reporting, updates the build to upload typescript maps, adds a Telemetry modal using cookies, and adds a error boundary in react to gather user feedback.

Closes #137

Some screenshots:
What the user sees:
Screen Shot 2019-11-25 at 8 48 58 PM
User feedback:
Screen Shot 2019-11-25 at 8 49 09 PM
Screen Shot 2019-11-25 at 8 51 00 PM
Telemetry Popup:
Screen Shot 2019-11-25 at 8 53 36 PM
Internal Error Report:
Screen Shot 2019-11-25 at 9 24 06 PM
Screen Shot 2019-11-25 at 9 23 15 PM

### CODE REVIEW

**Comment by braunsonm** [comment]
Other than tests, this is all done. For testing I'm not going to have the best luck in it. I might ask we delay it till after the feature freeze. Since this is error handling, it's going to be pretty hard to ensure an error handler is called since the error cannot be caught.

**Comment by braunsonm** [comment]
Manual tests `TI.1` and `TI.2` have been added to cover these functions!

File: src/components/TelemetryModal.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * with the users selection.
+ * @return {any} A react component which is the telemetry
+ * modal rendered
+ */
+export default function TelemetryModal() {
+  const cookies = new Cookies();
+  const prompt = cookies.get('telemetry');
+
+  // The assertion is handled here to check if the cookie has been set.
+  const [show, setShow] = useState(prompt === undefined);
```
Comment (camilleion) [inline src/components/TelemetryModal.tsx]: is this the right syntax? I thought assertions used the word 'assert'

File: src/components/TelemetryModal.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          button below.</p>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="danger" onClick={handleDisable}>
+            Disable Telemetry
+          </Button>
+          <Button variant="primary" onClick={handleClose}>
+            OK
+          </Button>
+        </Modal.Footer>
```
Comment (camilleion) [inline src/components/TelemetryModal.tsx]: I would add a comment or two for this chunk. I know it is a react component via the dock string but maybe a little more specifics like what is the modal?

File: src/components/TelemetryModal.tsx (hunk 1 above)
Comment (braunsonm) [reply src/components/TelemetryModal.tsx]: Heya, perhaps assertion isn't the right word. And maybe it's not a pre-condition. It's sort of an inbetween. Basically this dialog only appears if you don't have the cookie already, but the function checks that here. It's not necessarily an error state if the cookie exists, it just mean we don't need to render the popup.

**Comment by braunsonm** [comment]
@ets994 This needs your review as you are shadowing build master.

File: src/components/TelemetryModal.tsx (hunk 2 above)
Comment (mkoma808) [reply src/components/TelemetryModal.tsx]: A modal is an HTML popup like this https://www.w3schools.com/w3css/w3css_modal.asp

**Pull request merged**